src/Network/GRPC/LowLevel/Server.hs

```diff
@@ -154,14 +154,12 @@ addPort server conf@ServerConfig{..} =
            C.serverAddSecureHttp2Port server e creds
   where e = unEndpoint $ serverEndpoint conf
 
-startServer :: GRPC -> ServerConfig -> IO Server
+startServer :: GRPC -> ServerConfig -> IO (Server, Port)
 startServer grpc conf@ServerConfig{..} =
   C.withChannelArgs serverArgs $ \args -> do
     let e = serverEndpoint conf
     server <- C.grpcServerCreate args C.reserved
     actualPort <- addPort server conf
-    when (actualPort /= unPort port) $
```

Comment:
Why not keep this check, but slightly modified, for the case when `unPort port > 0` and isn't equal to `actualPort`?